ddtrace/tracer/span.go

```diff
@@ -583,6 +584,11 @@ func (s *Span) Finish(opts ...FinishOption) {
 	if s == nil {
 		return
 	}
+	// If Span.Finish has already been called, do nothing.
+	// In this way, we can ensure that Span.Finish is called at most once.
+	if !s.finishGuard.CompareAndSwap(0, 1) {
```

Comment:
Not a biggy, but using https://pkg.go.dev/sync/atomic#Bool for this could make this more readable.

Reply:
That's definitely acceptable compromise. Everything is an atomic int under the hood :D

Reply:
![image](https://github.com/user-attachments/assets/81da13d5-0b5b-468c-bc7a-7af2b835839b)